src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,240 @@
+					|| roadAccident.getRoadSurfaceConditions()
+							.equalsIgnoreCase("wet")) {
+				wet++;
+			} else if (roadAccident.getRoadSurfaceConditions().equals("5")
+					|| roadAccident.getRoadSurfaceConditions()
+							.equalsIgnoreCase("dry")) {
+				dry++;
+			}
+		}
+		map.put("Dry", dry);
+		map.put("Wet", wet);
+		return map;
+	}
+
+	/**
+	 * find the weather conditions which caused the top 3 number of incidents as
+	 * example if there were 10 accidence in rain, 5 in snow, 6 in sunny and 1
+	 * in foggy, then your result list should contain {rain, sunny, snow} - top
+	 * three in decreasing order
+	 * 
+	 * @return
+	 */
+	public List<String> getTopThreeWeatherCondition7() {
+		List<String> list = new LinkedList<>();
+		List<Long> countList = new ArrayList<>();
+		Multimap<String, Long> multiMap = HashMultimap.create();
+		Long count = 0L;
+
+		for (RoadAccident roadAccident : roadAccidentList) {
+			multiMap.put(roadAccident.getWeatherConditions(), count++);
+		}
+
+		for (int i = 0; i < 2; i++) {
+			countList.add(0L);
+		}
+
+		for (String key : multiMap.keySet()) {
+			long size = multiMap.get(key).size();
+			if (size > countList.get(0)) {
+				list.add(0, key);
+				countList.add(0, size);
+			} else if (size > countList.get(1)) {
+				list.add(1, key);
+				countList.add(1, size);
+			} else if (size > countList.get(2)) {
+				list.add(2, key);
+				countList.add(2, size);
+			}
+			if (list.size() == 4) {
+				list.remove(3);
+			}
+		}
+		return list;
+	}
+
+	/**
+	 * return a multimap where key is a district authority and values are
+	 * accident ids ex: authority1 -> id1, id2, id3 authority2 -> id4, id5
+	 * 
+	 * @return
+	 */
+	public Multimap<String, String> getAccidentIdsGroupedByAuthority7() {
+		Multimap<String, String> multiMap = HashMultimap.create();
+		long count = 0L;
+		for (RoadAccident roadAccident : roadAccidentList) {
+			multiMap.put(roadAccident.getDistrictAuthority(),
+					roadAccident.getAccidentId() + count++);
+		}
+		return multiMap;
+	}
+
+	// Now let's do same tasks but now with streaming api
+
+	public RoadAccident getAccidentByIndex(String index) {
+		RoadAccident roadAccident = roadAccidentList.stream()
+				.filter(ra -> ra.getAccidentId().equals(index)).findFirst()
+				.orElse(null);
+		return roadAccident;
+	}
+
+	/**
+	 * filter list by longtitude and latitude fields
+	 * 
+	 * @param minLongitude
+	 * @param maxLongitude
+	 * @param minLatitude
+	 * @param maxLatitude
+	 * @return
+	 */
+	public Collection<RoadAccident> getAccidentsByLocation(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		return roadAccidentList
+				.stream()
+				.filter(roadAccident -> roadAccident.getLongitude() >= minLongitude
+						&& roadAccident.getLongitude() <= maxLongitude)
+				.filter(roadAccident -> roadAccident.getLatitude() >= minLatitude
+						&& roadAccident.getLatitude() <= maxLatitude)
+				.collect(Collectors.toList());

```

Comment:
@zero1608 
for Collectors you can import it statically like import static java.util.stream.Collectors.*;